### ISSUE TITLE
feat: add a custom mirrors in docker/containerd config

### ISSUE THREAD

**Issue by winnieoursbrun** [aws/karpenter#3788]
### Tell us about your request

I'd like to had some mirrors in docker/containerd configuration.

Maybe the configuration should be in node-template like with kubelet ?

### Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?

Going through docker hub rate limiting, and maybe better performance. Also less network cost through NAT Gateway.
Actually i don't see anywhere where i can touch this kind of config with Karpenter.

### Are you currently working around this issue?

Nop.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by jonathan-innis** [comment]
Yep, makes sense. We talked about this a bit internally. Is there a way that you could update the docker config directly inside of the userData when the node starts up? I think you should be able to make additive or destructive changes to the configuration where these settings live on startup.

**Comment by winnieoursbrun** [comment]
Ah yes, i read it badly, i didn't know i could write bash script in userData. I'll try this way. Thanks.

**Comment by ivankatliarchuk** [comment]
Can someone share a working example for that?
